### ISSUE TITLE
Asset download fails due to exception trying to iterate over None

### ISSUE THREAD

**Issue by azurstarshine** [nobbyfix/AzurLane-AssetDownloader#1]
Set up:

1. Set `asset-directory` in the config file to my desired output location. (It's a directory at a higher level than where I have this repository checked out.)
1. Launch PowerShell and change the current directory to the local repository clone.
1. Create a virtual environment, update its pip/setuptools, and install required libraries:
    ```
    PS> py -m venv .venv
    PS> .\.venv\Scripts\Activate.ps1
    PS (.venv)> python -m pip install -U pip setuptools
    PS (.venv)> pip install -r .\requirements.txt
    ```
1. Change working directory to src directory: `PS (.venv)> cd .\src`.
1. Download the [EN XAPK file from ApkPure](https://apkpure.com/azur-lane-apk/com.YoStarEN.AzurLane).
1. Extract it as instructed:
    ```
    python .\obb_apk_import.py ..\ "..\..\Azur Lane_8.1.4_APKPure.xapk"
    ```

This works fine so far.

The problem:

1. Attempt to download assets using the command in the insutrctions:
    ```
    PS (.venv)> python .\main.py 'EN'
    ```

Result:

After downloading a bunch of files, it produces this error:

```
Downloading gallerypic/gallerypic55_t (107/110).
Downloading gallerypic/gallerypic6 (108/110).
Downloading gallerypic/gallerypic52 (109/110).
Downloading gallerypic/gallerypic49 (110/110).
BGM: Current version 1 is older than latest version 22.
Server returned empty hashfile for BGM, skipping.
Traceback (most recent call last):
  File "D:\az-work\AzurLane-AssetDownloader\src\main.py", line 52, in <module>
    main()
  File "D:\az-work\AzurLane-AssetDownloader\src\main.py", line 49, in main
    execute(args)
  File "D:\az-work\AzurLane-AssetDownloader\src\main.py", line 29, in execute
    update_assets = updater.update(vresult, clientconfig.cdnurl, userconfig, CLIENT_ASSET_DIR, args.force_refresh)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\az-work\AzurLane-AssetDownloader\src\lib\updater.py", line 132, in update
    return _update(version_result, cdnurl, userconfig, client_directory)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\az-work\AzurLane-AssetDownloader\src\lib\updater.py", line 122, in _update
    return _update_from_hashes(version_result, cdnurl, userconfig, client_directory, oldhashes, newhashes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\az-work\AzurLane-AssetDownloader\src\lib\updater.py", line 113, in _update_from_hashes
    comparison_results = compare_hashes(oldhashes, newhashes)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\az-work\AzurLane-AssetDownloader\src\lib\updater.py", line 36, in compare_hashes
    results = {row.filepath: CompareResult(None, row, CompareType.New) for row in newhashes}
                                                                                  ^^^^^^^^^
TypeError: 'NoneType' object is not iterable
```

(A small amount of preceding output is included to give the error some context.)

I've tried running this command twice, and the tool has done the exact same thing both times.

**Comment by azurstarshine** [comment]
(Forget that bit about not starting from where it failed. I had been messing with the config and had it set to a new directory. I thought I had reverted it before the second run, but I was wrong. I believe this is still a bug, though, since the output explicitly says that it's going to be *skipped*.)

**Comment by nobbyfix** [comment]
was fixed with 1b780ce a month ago
